elastic/kibana
# [Infra UI] Make KPI charts subtitle display the host limit information

closes https://github.com/elastic/kibana/issues/165825

## Summary

This ticket fixes the problem with the KPI subtitles that were not correctly showing the host limit information

https://github.com/elastic/kibana/assets/2767137/c2bb6109-82f6-4e2a-8a00-4ddd3f33fba5

Replacing the `useLazyRef` with `useMemo` in the `use_lens_attributes` to fix this problem, caused many problems with charts re-rendering, now that the attribute builder is no longer a singleton. 

Charts are heavy components and rendering them unnecessarily degrades performance considerably. So, besides fixing the main issue, I had to fix in this PR the other problems that surfaced after the change.

I also renamed a few things.

### How to test this PR
- Start a local Kibana instance
- Run slingshot with ~200 hosts for a couple of days
- Navigate to `Infrastructure` > `Hosts` and change the host limit to confirm that the KPI subtitle will reflect the selection
  -  Check if the clicks on the tabs and checkboxes are responding quickly.
